django/utils/synch.py

```diff
@@ -1,90 +0,0 @@
-"""
-Synchronization primitives:
-
-    - reader-writer lock (preference to writers)
-
-(Contributed to Django by eugene@lazutkin.com)
-"""
-
-import contextlib
-import threading
-
-
-class RWLock:
```

Comment:
wow, this is old, 12yo from 272eab5cd8ebafc68af3bca65f4f0d217c49ce53 . Doing a quick GitHub search I see one result in the first page that isn't a copy of Django, in django-request-cache: https://github.com/search?l=Python&q=django.utils.synch+RWLock&type=Code&utf8=%E2%9C%93 . I guess it's fine to remove though as it's undocumented, there are other options out there

Reply:
FYI, you can exclude filenames in github search - I find this very useful for excluding copies of Django code: https://github.com/search?utf8=%E2%9C%93&q=django.utils.synch+RWLock+-filename%3Alocmem.py+-filename%3Asynch.py+language%3APython&type=Code

Two pages of usages in a few modules, not too much.

Reply:
I tried with 'Path' on the advanced search dialog but it didn't seem to allow inverse, thanks for showing me the right syntax

@jarshwah one of the uses is actually in kogan/django-lrucache-backend 😉